### PR TITLE
Merge npm and yarn providers into node provider

### PR DESCRIPTION
**Pull request by wyzlle**

### CODE REVIEW

File: src/providers/node.rs [hunk 1]
```diff
@@ -36,26 +36,47 @@ impl Provider for NpmProvider {
...
+            if node_pkg.name != *DEFAULT_NODE_PKG_NAME {
+                yarn_pkg = yarn_pkg.set_override("nodejs", node_pkg.name.as_str());
+            }
+            return Ok(Some(SetupPhase::new(vec![node_pkg, yarn_pkg])));
+        }
 
         Ok(Some(SetupPhase::new(vec![node_pkg])))
     }
 
     fn install(&self, _app: &App, _env: &Environment) -> Result<Option<InstallPhase>> {
```
Comment (coffee-cup) [inline src/providers/node.rs:52]: Nit: change `_app` to `app`. Prefixed underscore should only be used if variable is unused

```suggestion
    fn install(&self, app: &App, _env: &Environment) -> Result<Option<InstallPhase>> {
```

**Review by coffee-cup — APPROVED** [review]
Nice! One small suggestion but looks good otherwise

File: src/providers/node.rs (hunk 1 above)
Comment (wyzlle) [reply src/providers/node.rs:52]: change `_app` to `app`

**Comment by coffee-cup** [comment]
Also in the future can you please add a PR description. Even something small is useful

**Comment by wyzlle** [comment]
> Also in the future can you please add a PR description. Even something small is useful

Sure, title was pretty self explanatory that's why I didn't add any description.

**Pull request merged**